### ISSUE TITLE
Incorrect E1029 errors when literal YAML block style is used

### ISSUE THREAD

**Issue by glerb** [aws-cloudformation/cfn-lint#2050]
*cfn-lint version: 0.51.0

Embedded parameters are being flagged for not being wrapped in a `!Sub`, but they are in fact wrapped in a `!Sub`.

Please provide as much information as possible:

Sample template (extraneous lines have been removed):
```yaml
Resources:
  SomeStackset:
    Type: AWS::CloudFormation::StackSet
    Properties:
      TemplateBody: |
        Resources:
          SomeRole:
            Type: AWS::IAM::Role
            Properties:
              Policies:
                -
                  PolicyName: SomeName
                  PolicyDocument:
                    Version: 2012-10-17
                    Statement:
                      -
                        Effect: Allow
                        Action:
                          - iam:GetSomething
                        Resource:
                          - !Sub arn:aws:iam::${AWS::AccountId}:role/SomeRole*
```

The error is an [E1029](https://github.com/aws-cloudformation/cfn-lint/blob/main/src/cfnlint/rules/functions/SubNeeded.py), and the line number referenced is the line number where the YAML literal appears (here, the line that contains `TemplateBody: |`)

This appears to be a new issue with [release 0.51.0](https://github.com/aws-cloudformation/cfn-lint/releases/tag/v0.51.0). The code in question above hasn't changed in my repo for a long time, and I don't get the error on previous releases.
